src/Drips.sol

```diff
@@ -359,34 +374,55 @@ abstract contract Drips {
+    /// The first item is always equal to `historyHash`.
+    /// @return currCycleConfigs The number of the sender's
+    /// drips configurations which have been seen in the current cycle.
+    /// This is also the number of used entries in each of the sender's `nextSqueezed` arrays.
     function _squeezeDripsResult(
         uint256 userId,
         uint256 assetId,
         uint256 senderId,
         bytes32 historyHash,
         DripsHistory[] memory dripsHistory
-    ) internal view returns (uint128 amt, uint256 squeezedNum, uint256[] memory squeezedIdxs) {
-        DripsState storage senderState = _dripsStorage().states[assetId][senderId];
-        _verifyDripsHistory(historyHash, dripsHistory, senderState.dripsHistoryHash);
+    )
+        internal
+        view
+        returns (
+            uint128 amt,
+            uint256 squeezedNum,
+            uint256[] memory squeezedRevIdxs,
+            bytes32[] memory historyHashes,
+            uint256 currCycleConfigs
+        )
+    {
+        {
+            DripsState storage sender = _dripsStorage().states[assetId][senderId];
+            historyHashes = _verifyDripsHistory(historyHash, dripsHistory, sender.dripsHistoryHash);
+            // If the last update was not in the current cycle,
+            // there's only the single latest history entry to squeeze in the current cycle.
+            currCycleConfigs = 1;
+            if (sender.updateTime >= _currCycleStart()) currCycleConfigs = sender.currCycleConfigs;
+        }
+        squeezedRevIdxs = new uint256[](dripsHistory.length);
         uint32[2 ** 32] storage nextSqueezed =
             _dripsStorage().states[assetId][userId].nextSqueezed[senderId];
         uint32 squeezeEnd = _currTimestamp();
-        // If the last update was not in the current cycle,
-        // there's only the single latest history entry to squeeze in the current cycle.
-        uint256 squeezedIdx = 1;
-        if (senderState.updateTime >= _currCycleStart()) squeezedIdx = senderState.currCycleConfigs;
-        uint256 historyIdx = dripsHistory.length;
-        squeezedIdxs = new uint256[](historyIdx);
-        while (squeezedIdx > 0 && historyIdx > 0) {
-            squeezedIdx--;
-            historyIdx--;
-            DripsHistory memory drips = dripsHistory[historyIdx];
+        for (uint256 i = 1; i <= dripsHistory.length && i <= currCycleConfigs; i++) {
+            DripsHistory memory drips = dripsHistory[dripsHistory.length - i];
             if (drips.receivers.length != 0) {
-                uint32 squeezeStart = nextSqueezed[squeezedIdx];
+                uint32 squeezeStart = nextSqueezed[currCycleConfigs - i];
                 if (squeezeStart < _currCycleStart()) squeezeStart = _currCycleStart();
                 if (squeezeStart < squeezeEnd) {
-                    squeezedIdxs[squeezedNum++] = squeezedIdx;
+                    squeezedRevIdxs[squeezedNum++] = i;
```

Comment:
It is a pity that the actual size of `squeezedRevIdxs` is unkown and we first need to iterate backwards at once to figure it out.

Outer-wise, we could have constructed it in the correct way.

Reply:
Maybe it make sense to to define the following variable

```solidity
historyHashIdx = currCycleConfigs - i;
```

```suggestion
                    squeezedRevIdxs[squeezedNum++] = historyHashIdx;
```

Reply:
Because the same subtraction happens in the higher function again

Reply:
Naming could be better because it is also the position in the `nextSqueezed[currCycleConfigs - i];`

Reply:
Ah true. You are right :+1: